module/os/linux/zfs/vdev_disk.c

```diff
@@ -659,8 +659,10 @@ __vdev_disk_physio(struct block_device *bdev, zio_t *zio,
 retry:
 	dr = vdev_disk_dio_alloc(bio_count);
 
-	if (zio && !(zio->io_flags & (ZIO_FLAG_IO_RETRY | ZIO_FLAG_TRYHARD)))
+	if (zio && !(zio->io_flags & (ZIO_FLAG_IO_RETRY | ZIO_FLAG_TRYHARD)) &&
+	    zio->io_vd->vdev_failfast == B_TRUE) {
 		bio_set_flags_failfast(bdev, &flags);
+	}
```

Comment:
I believe we wanted to add a module option to allow the `REQ_FAILFAST_MASK` bits to be modified.  Making the default value `REQ_FAILFAST_TRANSPORT | REQ_FAILFAST_DRIVER` (but not including `REQ_FAILFAST_DEV`) seems like a good default.  This way we retry everything but individual device errors, which I'd expect the device itself to have already handed internally.

Reply:
So just to be sure 100%.
We want per device option to disable/enable failfast?
And we want a module option to modify the mask?

Reply:
Yes, I think that's pretty reasonable.  We won't be able to control that mask on a per-device basis, but it keeps the interface easy to understand and as @allanjude mentioned in this https://github.com/openzfs/zfs/pull/14056#discussion_r1007330007 still provides us a mechanism to change the mask if needed.

Reply:
I agree this makes the most sense. Having it per-device was too much.